backend/app/resources/user_resources.py

```diff
@@ -0,0 +1,119 @@
+"""Definitions of user resources."""
+import base64
+from datetime import datetime
+from uuid import uuid4
+from bcrypt import gensalt, hashpw
+from flask_restful import Resource
+from flask import request
+from flask_jwt_extended import jwt_required, get_jwt_identity
+from ..models.user import AdminUser, EditorUser, RegularUser, User, UserSchema, RegularUserSchema, EditorUserSchema, AdminUserSchema
+from ..extensions import DB as db
+from ..utils.api_response import SuccessResponse, ErrorResponse
+from ..utils.decorators import roles_required
+
+USER_SCHEMA = UserSchema()
+USERS_SCHEMA = UserSchema(many=True)
+
+USER_CLASSES = {
+    'regular': RegularUser,
+    'admin': AdminUser,
+    'editor': EditorUser
+}
+
+USER_SCHEMAS = {
+    'regular': RegularUserSchema(),
+    'admin': AdminUserSchema(),
+    'editor': EditorUserSchema()
+}
+
+class UserListResource(Resource):
+    """Resource to handle listing users (admin only)."""
+
+    @jwt_required()
+    @roles_required("admin")
+    def get(self):
+        """Get a list of all users (admin only)."""
+        users = User.query.filter(User.deleted_at.is_(None)).all()
+        response = SuccessResponse(data=USERS_SCHEMA.dump(users))
+        return response.to_dict(), 200
+    
+    @jwt_required()
+    @roles_required("admin")
+    def post(self):
+        """Create a new user (admin only)."""
```

Comment:
For now let's leave this code as it's once you addressed current comments we will introduce a modular approach to simplify those "fat " methods. Personally, I don't like methods with more than 5 lines.